src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import core.basesyntax.service.Color;
+import core.basesyntax.service.ColorSupplier;
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public AbstractFigure getRandomFigure() {
+        Random random = new Random();
+        int figureType = random.nextInt(FigureType.values().length);
+
+        switch (figureType) {
+            case 0:
+                return new Square(random.nextDouble() * 10 + 1, colorSupplier.getRandomColor());
```

Comment:
let's extract this reapeated logic `random.nextDouble() * 10 + 1` to a private method, and preferably 10 to a constant